utilities/utility.py

```diff
@@ -14,6 +16,30 @@ def update_entity_fields(entity, **kwargs):
     return entity
 
 
+def cascade_soft_delete(parent_model, child, parent_id):
+    @event.listens_for(parent_model, 'after_update')
```

Comment:
@RonKbS is there a way we can have the two wrappers in one overall function, this way we can reduce complexity by reducing the number of functions.

Reply:
I have just pushed changes with one each decorator in its own function and this should solve the complexity issue @michael-basweti.

Thanks for the feedback.